eventsourcing/src/main/java/org/axonframework/eventsourcing/eventstore/SimpleTagResolver.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventsourcing.eventstore;
+
+import jakarta.annotation.Nonnull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static org.axonframework.common.BuilderUtils.assertNonNull;
+
+/**
+ * Simple implementation of the {@link TagResolver} that is able to combine several lambdas to construct
+ * {@link Tag Tags} from the event of type {@code E}.
+ * <p>
+ * The results of invoking all configured lambdas are combined into the {@link Set} of {@code Tags} when invoking
+ * {@link #resolve(Object)}.
+ *
+ * @param <E> The event for which to resolve a {@link Set} of {@link Tag Tags} for.
+ * @author Steven van Beelen
+ * @since 5.0.0
+ */
+public class SimpleTagResolver<E> implements TagResolver<E> {
+
+    private final List<Function<E, Tag>> tagResolvers;
+
+    private SimpleTagResolver(List<Function<E, Tag>> tagResolvers) {
+        this.tagResolvers = List.copyOf(tagResolvers);
+    }
+
+    /**
+     * Construct a {@code SimpleTagResolver} for the given {@code eventType}.
+     * <p>
+     * The initial resolver does not construct any {@link Tag Tags} yet. To add {@code Tags}, additional resolvers can
+     * be included through {@link #withResolver(Function)}.
+     *
+     * @param eventType The  {@code Class} of the event for which to resolve a {@link Set} of {@link Tag Tags} for.
+     * @param <E>       The event for which to resolve a {@link Set} of {@link Tag Tags} for.
+     * @return A {@code SimpleTagResolver} instance start, still requiring resolvers.
+     */
+    public static <E> SimpleTagResolver<E> forEvent(@SuppressWarnings("unused") Class<E> eventType) {
+        return new SimpleTagResolver<>(new ArrayList<>());
+    }
+
+    /**
+     * Construct a copy of {@code this SimpleTagResolver}, adding the given {@code tagResolver} to the set.
+     *
+     * @param tagResolver An additional {@code Function} from the event of type {@code E} to a {@link Tag}.
+     * @return A copy of {@code this SimpleTagResolver}, adding the given {@code tagResolver} to the set.
+     */
+    public SimpleTagResolver<E> withResolver(@Nonnull Function<E, Tag> tagResolver) {
```

Comment:
How do you decide what do you want to use as an argument? 
`TagResolver<E>` vs `Function<E, Tag>`?

As a rule, do you select the most generic types in case of arguments :) ?

Reply:
My frame of mind for this implementation, was that necessity for having an annotation-based format that can loop over all it's field to add separate lambdas for resolving a single `Tag`.
From that point of view, I assumed a implementation that:

1. Retrieves all fields of a class/record
2. Filters those fields for Axon-specific annotations
3. Map each applicable field to a `Tag`

In that flow, *we*, as providers of such an annotation-based version, will want singular `withResolver(Function<E,Tag>)` methods.
Whether a user will use this? Maybe, maybe not. At least, they don't have to. They can pass a lamdba directly, if desired, to the overall configuration API.